server/app/mutations/grid_services/update.rb

```diff
@@ -79,6 +86,9 @@ def execute
       if self.secrets
         attributes[:secrets] = self.build_grid_service_secrets(self.grid_service.secrets.to_a)
       end
+      if self.volumes
+        attributes[:service_volumes] = self.build_service_volumes
```

Comment:
Needs to handle validation errors.

Best to raise an error, and then have a def-level `except` that does `add_error`...